zipline/pipeline/loaders/blaze/core.py

```diff
@@ -1011,12 +1011,47 @@ def last_in_date_group(df, reindex, have_sids=have_sids):
                     )
                 else:
                     last_in_group = last_in_group.reindex(dates)
+            # Unstack will fill all missing values with NaN; we need to fix
+            # this for all types that are not float.
+            if not df.empty:
+                for column in columns:
+                    if isinstance(column.missing_value, NotSpecifiedType):
+                        raise ValueError("Must specify a missing value for "
+                                         "column %d." % column.name)
+                    # Need special logic in case we're replacing with Nones
+                    elif df[column.name].dtype == categorical_dtype:
+                        last_in_group[column.name] = last_in_group[
+                            column.name
+                        ].where(pd.notnull(last_in_group[column.name]),
+                                column.missing_value)
+                    # Need to convert from float col to datetime col
+                    elif df[column.name].dtype == datetime64ns_dtype:
```

Comment:
Yeah, good catch, I think that was there twice by accident. I also don't think I need that `fillna` there. 

I was doing it in the datetime64 case because it's the only way to fill in `NaT` for null values. We could do it for the others, but it didn't seem necessary since the actual output of this method is already doing that conversion for each column/column dtype.